test/create_examples_emr_test.go

```diff
@@ -0,0 +1,137 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func initTestCases() []EmrTestCase {
+	return []EmrTestCase{
+		{
+			testName: "Hbase_Test_1",
+			vars: map[string]interface{}{
+				"vpc_cidr":            "172.22.0.0/20",
+				"private_subnets":     []string{"172.22.1.0/24", "172.22.2.0/24"},
+				"public_subnets":      []string{"172.22.3.0/24", "172.22.4.0/24"},
+				"ingress_cidr_blocks": []string{"0.0.0.0/0"},
+				"egress_cidr_blocks":  []string{"0.0.0.0/0"},
+				"tags":                make(map[string]string),
+				"abac_tags": map[string]string{
+					"for-use-with-amazon-emr-managed-policies": "true",
+				},
+				"abac_valid_tags": make(map[string][]string),
+			},
+			expectApplyError: false,
+			tfDir:            "test_examples/static-hbase",
+		},
+		{
+			testName: "Spark_Test_1",
+			vars: map[string]interface{}{
+				"vpc_cidr":            "172.23.0.0/20",
+				"private_subnets":     []string{"172.23.1.0/24", "172.23.2.0/24"},
+				"public_subnets":      []string{"172.23.3.0/24", "172.23.4.0/24"},
+				"ingress_cidr_blocks": []string{"0.0.0.0/0"},
+				"egress_cidr_blocks":  []string{"0.0.0.0/0"},
+				"tags":                make(map[string]string),
+				"abac_tags": map[string]string{
+					"for-use-with-amazon-emr-managed-policies": "true",
+				},
+				"abac_valid_tags": make(map[string][]string),
+			},
+			expectApplyError: false,
+			tfDir:            "test_examples/static-spark",
+		},
+		{
+			testName: "Ephem_Spark_Test_1",
+			vars: map[string]interface{}{
+				"vpc_cidr":       "172.24.0.0/20",
+				"public_subnets": []string{"172.24.3.0/24", "172.24.4.0/24"},
+				"tags":           make(map[string]string),
+				"abac_tags": map[string]string{
+					"for-use-with-amazon-emr-managed-policies": "true",
+				},
+				"abac_valid_tags": make(map[string][]string),
+			},
+			expectApplyError: false,
+			tfDir:            "test_examples/ephemeral-spark",
+		},
+	}
+}
+func TestCreateExamplesEmr(t *testing.T) {
+	// os.Setenv("TERRATEST_REGION", "us-east-1")
+
+	testCases := initTestCases()
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.testName, func(t *testing.T) {
+			t.Parallel()
+			// These will create a tempTestFolder for each bucketTestCase.
+			tempTestFolder := test_structure.CopyTerraformFolderToTemp(t, "..", testCase.tfDir)
+
+			// this stage will generate a random `awsRegion` and a `uniqueId` to be used in tests.
+			test_structure.RunTestStage(t, "pick_new_randoms", func() {
+				// Pick a random AWS region to test in. This helps ensure your code works in all regions.
+				usRegions := []string{"us-east-1", "us-east-2", "us-west-1", "us-west-2"}
```

Comment:
Shouldn't we abstract this configuration to make it easier to change all regions from a single line?

Reply:
`os.Setenv("TERRATEST_REGION", "us-east-1")` would do that for you if you need/want. And I see it being only useful for local testing.